### PR TITLE
Fix clippy lints

### PR DESCRIPTION
**Pull request by brendanzab**
Thought it was about time I fixed these clippy lints.

Seemed to uncover a bunch of reasonable issues even the there are some lints I’m a bit wary of.

### CODE REVIEW

File: fathom/src/core/semantics.rs [hunk 1]
```diff
@@ -173,6 +174,7 @@ impl<'arena> Telescope<'arena> {
     }
 
     /// The number of terms in the telescope.
+    #[allow(clippy::len_without_is_empty)]
```
Comment (wezm) [inline fathom/src/core/semantics.rs:177]: Maybe just turn this one off? Can't ever seeing it adding much value.

File: fathom/src/core/semantics.rs [hunk 2]
```diff
@@ -164,6 +164,7 @@ impl<'arena> Telescope<'arena> {
         }
     }
 
+    #[allow(clippy::bool_assert_comparison)]
```
Comment (wezm) [inline fathom/src/core/semantics.rs:167]: Maybe turn this one off too. Seems low-value.

**Review by wezm — APPROVED** [review]
Overall good improvements. I definitely agree there's some lints of dubious value that we might choose to disable.

**Comment by brendanzab** [comment]
Think I’m going to enable clippy a separate PR. Some shenanigans is going on! 🤔

**Pull request merged**